src/commandline_options.cpp

```diff
@@ -205,9 +205,9 @@ commandline_options::commandline_options(const std::vector<std::string>& args)
 		("unsafe-scripts", "makes the \'package\' package available to Lua scripts, so that they can load arbitrary packages. Do not do this with untrusted scripts! This action gives ua the same permissions as the Wesnoth executable.")
 		("usercache-dir", po::value<std::string>(), "sets the path of the cache directory to $HOME/<arg> or My Documents\\My Games\\<arg> for Windows. You can specify also an absolute path outside the $HOME or My Documents\\My Games directory. Defaults to $HOME/.cache/wesnoth on X11 and to the userdata-dir on other systems.")
 		("usercache-path", "prints the path of the cache directory and exits.")
-		("userconfig-dir", po::value<std::string>(), "sets the path of the user config directory to $HOME/<arg> or My Documents\\My Games\\<arg> for Windows. You can specify also an absolute path outside the $HOME or My Documents\\My Games directory. Defaults to the userdata-dir.")
+		("userconfig-dir", po::value<std::string>(), "sets the path of the user config directory. You must specify an absolute path outside the $HOME or My Documents\\My Games directory. Defaults to the userdata-dir.")
```

Comment:
Where is this requirement coming from?

Reply:
That's... what this PR is removing? The previously deprecated behavior was to default to the home or My Games directory when a relative path was provided.

Reply:
Maybe I'm missing something but nothing changes for the user config dir as far as I can tell.

Reply:
I think I'm confused as well. The existing text seems just wrong? Since the user config directory setter has never had the same logic as the userdata directory setup to make it be relative to the home directory.

Reply:
That's true nothing special is done so the part about $HOME and My Games etc is just wrong.

Reply:
So... I think I changed this text for the wrong reason, but it ended up being the right outcome anyway since the existing text was already wrong.

Reply:
`You must specify an absolute path.` is incorrect though.

Reply:
Removed that sentence.